### PR TITLE
W-10224258 - Add data card component

### PR DESCRIPTION
**Pull request by TidharPeer**
## Description
Write here a short description

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Are you adding a new package?
- [ ] yes
- [x] no

image
```
CTC Output:
Case approved, merging is possible.
The case is available at: https://gus.lightning.force.com/lightning/r/Case/500EE00000FPTarYAH/view
Updated at Tue Dec 07 09:58:20 UTC 2021
```

### CODE REVIEW

File: src/components/DataCard.js [hunk 1]
```diff
@@ -0,0 +1,232 @@
+/* eslint react/prop-types: 0 */
+import React, { useState } from 'react';
+import styled, { css } from 'styled-components';
+import * as PropTypes from 'prop-types';
+import { Checkbox } from './Checkbox';
+import { ReactComponent as ChevronUpDoubleAsset } from '../assets/chevron-up-double.svg';
```
Comment (kaufguy) [inline src/components/DataCard.js:6]: pleaes use absolute path

File: src/components/DataCard.js [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+  .icon-container {
+    width: 16px;
+    height: 16px;
+    img {
+      width: 16px;
+      height: 16px;
+    }
+    &.icon-background {
+      background: ${({ theme }) => theme.p0};
+      box-shadow: 0px 0px 8px rgba(79, 80, 83, 0.15);
```
Comment (kaufguy) [inline src/components/DataCard.js:120]: theme?

File: src/components/DataCard.js [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+`;
+
+const CardContainer = styled.div`
+  position: relative;
+  border-radius: 2px;
+  box-shadow: 0 4px 23px 0 rgba(0, 0, 0, 0.1);
+  padding: 12px 17px 15px;
+  display: flex;
+  flex-direction: column;
+  border: 1px solid rgba(0, 105, 126, 0.15);
```
Comment (kaufguy) [inline src/components/DataCard.js:142]: theme?

File: src/components/DataCard.js [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+  const [selected, setSelected] = useState(props.checked);
+  const [isMoreOpen, setIsMoreOpen] = useState(false);
+  return (
+    <Container
+      width={props.width}
+      height={props.height}
+      className={`data-card ${props.className ? props.className : ''}`}
+    >
+      <CardContainer
+        onClick={() => setSelected(!selected)}
```
Comment (kaufguy) [inline src/components/DataCard.js:19]: I dont think the click should be on the container but on the checkbox.
Also, I think onClick should be a property that is controlled from the outside (and will modify the checked prop that comes from outside). No need for selected state (its just the 'checked' prop)

File: src/components/DataCard.js (hunk 1 above)
Comment (TidharPeer) [reply src/components/DataCard.js:6]: Changed to this:
image

File: src/components/DataCard.js (hunk 4 above)
Comment (TidharPeer) [reply src/components/DataCard.js]: Done

File: src/components/DataCard.js [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+          <div
+            className={'more-less-holder'}
+            onClick={() => setIsMoreOpen(!isMoreOpen)}
+          >
+            {isMoreOpen ? (
+              <div className={'d-flex'}>
+                <span className={'mr-5'}>{props.lessLabel || 'Less'}</span>
+                <ChevronUpDoubleAsset
+                  className={'primary-300-color'}
+                  width={'8px'}
```
Comment (distroysf) [inline src/components/DataCard.js:59]: put the width in ChevronUpDoubleAsset styled component

File: src/components/DataCard.js [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+
+export const DataCard = (props) => {
+  const [isMoreOpen, setIsMoreOpen] = useState(false);
+  return (
+    <Container
+      className={`data-card ${props.className ? props.className : ''}`}
+    >
+      <CardContainer
+        onClick={props.onClick}
+        className={`card-container ${props.checked ? 'selected' : ''}`}
```
Comment (distroysf) [inline src/components/DataCard.js:19]: no need for classes for every component

**Review by distroysf — APPROVED** [review]

**Review by distroysf — CHANGES_REQUESTED** [review]
remove use of classNames

File: src/components/DataCard.js [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+      >
+        <Header>
+          {props.icon && (
+            <IconContainer background={props.iconBackground}>
```
Comment (distroysf) [inline src/components/DataCard.js:23]: background prop is not used

File: src/components/DataCard.js [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+            </IconContainer>
+          )}
+          <CompanyName>{props.name}</CompanyName>
+        </Header>
+        <Checkbox className={'card-checkbox'} checked={props.checked} />
+        {props.subHeader && <SubHeader>{props.subHeader}</SubHeader>}
+        {props.children}
+      </CardContainer>
+      {props.hasMoreInfo && (
+        <MoreContainer height={props.height} borderColor={props.borderColor}>
```
Comment (distroysf) [inline src/components/DataCard.js:34]: remove all style attrs from html

File: src/components/DataCard.js [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+          {isMoreOpen && (
+            <MoreChildrenContainer>
+              {React.cloneElement(props.moreChildren)}
+            </MoreChildrenContainer>
+          )}
+          <MoreLessHolder onClick={() => setIsMoreOpen(!isMoreOpen)}>
+            {isMoreOpen ? (
+              <ChevronUpContainer>
+                <Label>{props.lessLabel || 'Less'}</Label>
```
Comment (distroysf) [inline src/components/DataCard.js:43]: for hardcoded strings use `useTranslations`

File: src/components/DataCard.js [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+        </Header>
+        <Checkbox className={'card-checkbox'} checked={props.checked} />
+        {props.subHeader && <SubHeader>{props.subHeader}</SubHeader>}
+        {props.children}
+      </CardContainer>
+      {props.hasMoreInfo && (
+        <MoreContainer height={props.height} borderColor={props.borderColor}>
+          {isMoreOpen && (
+            <MoreChildrenContainer>
+              {React.cloneElement(props.moreChildren)}
```
Comment (distroysf) [inline src/components/DataCard.js:37]: how it's used moreChildren

File: src/components/DataCard.js [hunk 11]
```diff
@@ -0,0 +1,225 @@
...
+      >
+        <Header>
+          {props.icon && (
+            <IconContainer background={props.iconBackground}>
+              <img src={props.icon} />
+            </IconContainer>
+          )}
+          <CompanyName>{props.name}</CompanyName>
+        </Header>
+        <Checkbox className={'card-checkbox'} checked={props.checked} />
```
Comment (distroysf) [inline src/components/DataCard.js:29]: remove className={'card-checkbox'}

File: src/components/DataCard.js (hunk 10 above)
Comment (TidharPeer) [reply src/components/DataCard.js:32]: Its the component that being rendered inside the More, if its open

File: src/components/DataCard.js (hunk 9 above)
Comment (TidharPeer) [reply src/components/DataCard.js:38]: Didn't see a single use for it in the app components... this is why I added the props, so users can add their translations

Comment (distroysf) [reply src/components/DataCard.js:38]: you're correct

File: src/components/DataCard.js [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+import { Checkbox } from './Checkbox';
+import { ReactComponent as ChevronUpDoubleAsset } from '/src/assets/chevron-up-double.svg';
+import { ReactComponent as ChevronDownDoubleAsset } from '/src/assets/chevron-down-double.svg';
+import { Card } from './Card';
+import { hexToRgba } from '../utils/theme.utils';
+
+export const DataCard = (props) => {
+  const [isMoreOpen, setIsMoreOpen] = useState(false);
+  return (
+    <Container className={`${props.className ? props.className : ''}`}>
```
Comment (distroysf) [inline src/components/DataCard.js:14]: remove class

File: src/components/DataCard.js [hunk 13]
```diff
@@ -0,0 +1,223 @@
...
+  border-radius: 8px;
+  box-shadow: 0 4px 23px 0 ${({ theme }) => hexToRgba(theme.p700, 10)};
+  padding: 12px 17px 15px;
+  display: flex;
+  flex-direction: column;
+  z-index: 1;
+  border: 1px solid transparent;
+  border-bottom-color: transparent;
+  cursor: pointer;
+  ${({ height }) =>
```
Comment (distroysf) [inline src/components/DataCard.js:117]: there isn't height prop

**Review by distroysf — CHANGES_REQUESTED** [review]

File: src/components/DataCard.js [hunk 14]
```diff
@@ -0,0 +1,223 @@
...
+  margin-top: 8px;
+  margin-bottom: 2px;
+  height: 14px;
+  font-weight: 600;
+  font-size: 10px;
+  color: ${({ theme }) => theme.p300};
+`;
+
+const HeaderLabel = styled.span`
+  margin-left: 7px;
```
Comment (distroysf) [inline src/components/DataCard.js:159]: remove all custom margins
this won't grow well in different sizes

**Review by distroysf — APPROVED** [review]

**Review by kaufguy — APPROVED** [review]

**Pull request merged**